src/main/resources/documentation/domain-model-overview.md

```diff
@@ -0,0 +1,257 @@
+######Required Fields:
++ int keyIdentifier
++ String name
+
+___
+####Location
+(child of `AuditableExtIdEntity`)
+#####What is it?
+`Location` is the leaf node on the `LocationHierarchy` tree, it is the smallest geographical unit within a study area. From the previous example for `LocationHierarchy`, each building within a district could be modeled with `Location`. `Individual`s are residents at `Locations`.
+######Dependencies
++ FieldWorker
++ LocationHierarchy
++ Domain Constraints
+
+######Required Fields:
++ String name
++ String extId
++ LocationHierarchy parent
+
+___
+####Individual
+(child of `AuditableExtIdEntity`)
+#####What is it?
+An `Individual` (person) within the study area that resides at a `Location`, is a member of a `SocialGroup`, and has a relationship to the head of a household.
+######Dependencies
++ FieldWorker
+
+######Required Fields:
++ String extId
++ String gender
++ Date dateOfBirth
++ String firstName
+
+___
+####SocialGroup
+(child of `AuditableExtIdEntity`)
+#####What is it?
+A `SocialGroup` is any cultural or societal group or collective that `Individual`s are a member of, e.g. a church group, a household/family, a bowling team.
+######Dependencies
++ FieldWorker
+
+######Required Fields:
++ String groupName
++ String groupType
++ String extId
+
+___
+###Membership
+(child of `AuditableCollectedEntity`)
+####What is it?
+A `Membership` models the relationship between an `Individual` and a `SocialGroup`.
+######Dependencies
++ FieldWorker
++ Individual
++ SocialGroup
+
+######Required Fields:
++ Date startDate
++ String startType
++ Individual individual
++ SocialGroup socialGroup
+
+___
+####Relationship
+(child of `AuditableCollectedEntity`)
+#####What is it?
+A `Relationship` models the relationship between two `Individual`s. Specifically it models the relationship between an `Individual` and the head of household that the `Individual` is a part of.
+######Dependencies
++ FieldWorker
++ Individual
+
+######Required Fields:
++ Date startDate
++ String relationshipType
++ Individual individualA
++ Individual individualB
+
+___
+####Residency
+(child of `AuditableCollectedEntity`)
+#####What is it?
+A `Residency` models the relationship between an `Individual` and a `Location`.
+######Dependencies
++ FieldWorker 
++ Individual
++ Location
+
+######Required Fields:
++ Date startDate
++ String relationshipType
++ Individual individualA
++ Individual individualB
+
+___
+#Update Entities
+####Visit
+(child of `AuditableExtIdEntity`)
+#####What is it?
+A `Visit` models a `FieldWorker`’s follow-up visit to a particular `Location` at a particular date. `Visit`s are necessary for all “update” events that can take place within a population.
+######Dependencies
++ FieldWorker
++ Location 
+
+######Required Fields:
++ Date visitDate
++ Location location
+
+___
+####Death
+(child of `AuditableCollectedEntity`)
+#####What is it?
+A `Death` models the death of one of the `Individual`s within the area of study.

```

Comment:
It might be good to use the language of the reference data model paper here.  Instead of saying Death models something, you could say that Death is a record of an event.